pkg/util/conditions.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	gcpv1alpha1 "github.com/openshift/gcp-project-operator/pkg/apis/gcp/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetProjectClaimCondition sets a condition on a ProjectClaim resource's status
+func SetProjectClaimCondition(
+	conditions []gcpv1alpha1.ProjectClaimCondition,
+	conditionType gcpv1alpha1.ProjectClaimConditionType,
+	status corev1.ConditionStatus,
+	reason string,
+	message string,
+) []gcpv1alpha1.ProjectClaimCondition {
+	now := metav1.Now()
+	existingCondition := FindProjectClaimCondition(conditions, conditionType)
+	if existingCondition == nil {
+		if status == corev1.ConditionTrue {
+			conditions = append(
+				conditions,
+				gcpv1alpha1.ProjectClaimCondition{
+					Type:               conditionType,
+					Status:             status,
+					Reason:             reason,
+					Message:            message,
+					LastTransitionTime: now,
+					LastProbeTime:      now,
+				},
+			)
+		}
+	} else {
+		// If it does not exist, assign it as now. Otherwise, do not touch
+		if existingCondition.Status != status {
+			existingCondition.LastTransitionTime = now
+		}
+		existingCondition.Status = status
+		existingCondition.Reason = reason
+		existingCondition.Message = message
+		existingCondition.LastProbeTime = now
+	}
+	return conditions
+}
+
+// FindProjectClaimCondition finds in the condition that has the
+// specified condition type in the given list. If none exists, then returns nil.
+func FindProjectClaimCondition(conditions []gcpv1alpha1.ProjectClaimCondition, conditionType gcpv1alpha1.ProjectClaimConditionType) *gcpv1alpha1.ProjectClaimCondition {
+	for i, condition := range conditions {
+		if condition.Type == conditionType {
+			return &conditions[i]
+		}
+	}
+
+	return nil
+}
```

Comment:
I think those should go into the CustomResourceAdapter. You could create the adapter in the Reconcile function and pass it on to the ReconcileHandler instead of the ProjectClaim I think.

Reply:
done